### DISCUSSION TITLE
I want to transcribe audio in real-time concurrently with recording using PyAudio's stream

### DISCUSSION THREAD

**Discussion by houkagoplay** [SYSTRAN/faster-whisper#929] [Q&A]
"I'm currently developing a program using PyAudio to record microphone audio and perform real-time speech-to-text transcription. I've successfully saved the audio data as a file using stream.read, but I'm struggling to perform the transcription. Instead of transcribing the saved file immediately after recording, I want to transcribe it concurrently with the recording process. However, I'm unsure how to pass the audio data from the streaming process to the transcription process.

In the example provided by the library:

> segments, info = model.transcribe("audio.mp3", beam_size=5)

Is it possible to use the audio data captured during recording in place of "audio.mp3"? If so, what processing steps are needed to pass the streaming data for transcription?

I would appreciate any guidance or suggestions on how to achieve this.

This document was created using Google Translate and chatgpt.

**Comment by MahmoudAshraf97** [comment]
You can pass a `float32` array sampled at 16khz

**Reply by houkagoplay** [reply]
Thank you!! I will try that!!

**Reply by SlareJ** [reply]
Did it work? Facing the same issue, but this one doesn't seem to help

**Reply by rabyunghwa** [reply]
@SlareJ It didn't work for me, either. What error did you have? Mine was:

> <built-in method with_traceback of InvalidDataError object at 0x7b251c541f60>

**Reply by SlareJ** [reply]
@rabyunghwa I don't remember encountering an error, it just didn't work for me. The transcription was blank

**Reply by rabyunghwa** [reply]
@SlareJ Okay. No solution yet?

**Comment by trungkienbkhn** [comment]
You can try using the 'sounddevice' lib:
```python
import numpy as np
import sounddevice as sd

from faster_whisper import WhisperModel

print("Recording started")
duration = 10
sample_rate = 16000
audio_data = sd.rec(
    int(sample_rate * duration), samplerate=sample_rate, channels=1, dtype=np.float32
)
sd.wait()
audio_data = audio_data.squeeze()
print(audio_data)
print("Recording stopped")

model = WhisperModel("tiny", device="cpu")
segments, info = model.transcribe(audio_data, word_timestamps=True)
for segment in segments:
    print("[%.2fs -> %.2fs] %s" % (segment.start, segment.end, segment.text))
```

**Reply by houkagoplay** [reply]
Thank you! I'll try this method too!

**Reply by rabyunghwa** [reply]
@trungkienbkhn Any idea why 'sounddevice' lib works whereas PyAudio doesn't?
